plugins/woocommerce/includes/admin/reports/class-wc-report-taxes-by-code.php

```diff
@@ -208,8 +208,31 @@ public function get_main_chart() {
 						$rate = $wpdb->get_var( $wpdb->prepare( "SELECT tax_rate FROM {$wpdb->prefix}woocommerce_tax_rates WHERE tax_rate_id = %d;", $rate_id ) );
 						?>
 						<tr>
-							<th scope="row"><?php echo wp_kses_post( apply_filters( 'woocommerce_reports_taxes_tax_rate', $tax_row->tax_rate, $rate_id, $tax_row ) ); ?></th>
-							<td><?php echo wp_kses_post( apply_filters( 'woocommerce_reports_taxes_rate', $rate, $rate_id, $tax_row ) ); ?>%</td>
+							<th scope="row">
+								<?php
+									echo wp_kses_post(
+										apply_filters_deprecated(
+											'woocommerce_reports_taxes_tax_rate',
+											array( $tax_row->tax_rate, $rate_id, $tax_row ),
+											'9.5.0',
+											null,
+											'Reports are deprecated and will be removed in future versions. Use Analytics instead.',
+										)
+									);
+								?>
+							</th>
+							<td>
+								<?php
+									echo wp_kses_post(
+										apply_filters_deprecated(
+											'woocommerce_reports_taxes_rate',
+											array( $rate, $rate_id, $tax_row ),
+											'9.5.0',
+											null,
+											'Reports are deprecated and will be removed in future versions. Use Analytics instead.',
+										)
+									);
```

Comment:
As mentioned above, should we ensure `wp_kses_post` is never passed a null?

```suggestion
									echo wp_kses_post(
										apply_filters_deprecated(
											'woocommerce_reports_taxes_rate',
											array( $rate, $rate_id, $tax_row ),
											'9.5.0',
											null,
											'Reports are deprecated and will be removed in future versions. Use Analytics instead.',
										) ?? ''
									);
```